### ISSUE TITLE
 Better a11y: correct aria-modal usage for Datepicker's Dropdown 

### ISSUE THREAD

**Issue by mateuswetah** [buefy/buefy#3683]
### Overview of the problem

I've been doing some tests with the [Axe Dev Tools](https://www.deque.com/axe/devtools/) extension. It does a pretty good job on pointing accessibility errors. While I think Buefy has improved enormously in the last years regarding this topic, the reports shows that there are some misuse of Aria attributes, so we have some work to do! This is the second of a series of small issues that I am raising to cover the ones that I understand. I am by no means an expect on the topic, so would appreciate reviews and comments.

**Buefy** version: 0.9.17
**Vuejs** version: 2.6.3
**OS/Browser**: Chrome/Linux

### Description

The Datepicker component has a Dropdown that is receiving a `aria-modal="true"` around when the component is not inline. This may sound correct, but the attribute should be passed to the same element that has a `dialog` or `modal` role, which in this case is the `.dropdown-menu`.

### Steps to reproduce
1. Install the [Axe Dev Tools](https://www.deque.com/axe/devtools/) extension. Sadly, only for Chrome...
2. Run it on the documentation page about the Datepicker component: https://buefy.org/documentation/datepicker
3. Check the issues panel. There shall be a series of ones regarding **Elements must only use allowed ARIA attributes**

![image](https://user-images.githubusercontent.com/1184874/155545562-4b90f078-2eec-4130-b075-29dacd247e63.png)

### Expected behavior

As suggested, to remove the  `aria-modal="true"` from this element, and possibly move it to the `.dropdown-menu`. Should be a simple pull request ;)
